src/content/en/fundamentals/instant-and-offline/offline-storage-pwa/index.md

```diff
@@ -0,0 +1,260 @@
+
+## **What about other storage mechanisms?**
+
+* Web Storage (e.g LocalStorage and SessionStorage) is synchronous, has no Web
+Worker support and is size-limited (only strings).
+
+* Cookies [have their
+uses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) but are
+synchronous, lack Web Worker support and are also size-limited.
+
+* WebSQL does not have broad browser support and its use is not recommended.
+
+* File System API is not supported on any browser besides Chrome.
+
+* The [File API](https://developer.mozilla.org/en-US/docs/Web/API/File) is
+being improved in the [File and Directory Entries
+API](https://wicg.github.io/entries-api/) and [File
+API](https://w3c.github.io/FileAPI/) specs but neither is sufficiently mature
+or standardized to encourage widespread adoption yet.
+
+## **Some helpful resources**
+
+### **IndexedDB Libraries worth checking out:**
+
+* [localForage](https://mozilla.github.io/localForage/)(~8KB, Promises, good
+legacy browser support)
+
+* [Dexie](http://dexie.org/) (~16KB, Promises, complex queries, secondary
+indices)
+
+* [PouchDB](https://pouchdb.com/) (~45KB (supports [custom
+builds](https://pouchdb.com/2016/06/06/introducing-pouchdb-custom-builds.html)),
+synchronization)
+
+* [Lovefield](https://github.com/google/lovefield)(relational)
+
+* [LokiJS](http://lokijs.org/#/) (in-memory)
+
+* [ydn-db](https://github.com/yathit/ydn-db)(dexie-like, works with WebSQL)
+
+### **Service Worker Libraries worth checking out:**
+
+* [sw-toolbox](https://github.com/GoogleChrome/sw-toolbox) (offline caching
+for dynamic/runtime requests)
+
+* [sw-precache](https://github.com/GoogleChrome/sw-precache) (offline
+precaching for static assets/application shells)
+
+* Webpack users can directly use the above or
+[offline-plugin](https://github.com/NekR/offline-plugin)
+
+## **How much can I store?**
+
+In Chrome and Opera, your storage is per origin (rather than per API). Both
+storage mechanisms will store data until the browser
+[quota](http://www.html5rocks.com/en/tutorials/offline/quota-research/) is
+reached. Apps can check how much quota they’re using with the [Quota Management
+API](https://www.w3.org/TR/quota-api/). In Firefox, apps can use up to 10% of
+free disk space, but will prompt after 50MB data stored. In mobile Safari, apps
+can use up to 50MB max, whereas desktop Safari allows unlimited storage (and
+prompts after 5MB). IE10+ maxes out at 250MB and prompts at 10MB. PouchDB
+[tracks](https://pouchdb.com/faq.html#data_limits) IDB storage behavior.
+
+## **How can I tell how much storage space my app is using?**
+
+The [Quota Management API](https://www.w3.org/TR/quota-api/) lets you query for
+the size of storage space currently used and how much is available to the
+application. It also enables requesting for more storage if needed.  A newer
+[Storage Quota Estimate
+API](https://www.chromestatus.com/features/5630353511284736) tries to make it
```

Comment:
" It also enables requesting for more storage if needed." - we really shouldn't call this out. It applies only to the PERSISTENT bucket which is only used by the FileSystem API. Without those caveats it is misleading.